test/Microsoft.ML.Functional.Tests/Prediction.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.SamplesUtils;
+using Microsoft.ML.TestFramework;
+using Xunit;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public class PredictionScenarios
+    {
+        /// <summary>
+        /// Reconfigurable predictions: The following should be possible: A user trains a binary classifier,
+        /// and through the test evaluator gets a PR curve, the based on the PR curve picks a new threshold
+        /// and configures the scorer (or more precisely instantiates a new scorer over the same predictor)
+        /// with some threshold derived from that.
+        /// </summary>
+        [Fact]
+        public void ReconfigurablePrediction()
+        {
+            var mlContext = new MLContext(seed: 789);
+
+            // Get the dataset, create a train and test
+            var dataset = DatasetUtils.LoadHousingRegressionDataset(mlContext, BaseTestClass.GetDataPath("housing.txt"));
```

Comment:
>LoadHousingRegressionDataset [](start = 39, length = 28)

just reference the housing dataset like the other tests are doing. Let's leave DatasetUtils for the samples.

Reply:
oh I see, you are defining the TextLoader etc in there. I believe there is a separate test file for those, in TestDataets. 

---
In reply to: [255278973](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255278973) [](ancestors = 255278973)

Reply:
Good point. I went ahead and added this just like the other tests do. It looks like we'll need to do some refactoring so that we can push most of the loader and options into the dataset object as well, but that can wait for later.

---
In reply to: [255279066](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255279066) [](ancestors = 255279066,255278973)